File: tacas2020-benchmarks/booster/sina5.c
#define N

int main()
{
	int i;
	int sum;
	int a[N];
	int b[N];

	sum = 0;
	for(i=0; i<N; i++)
	{
		a[i] = 1;
	}

	for(i=0; i<N; i++)
	{
		b[i] = 1;
	}

	for(i=0; i<N; i++)
	{
		sum = sum + a[i];
	}

	for(i=0; i<N; i++)
	{
		sum = sum + b[i];
	}

	for(i=0; i<N; i++)
	{
		a[i] = a[i] + sum;
	}

	for(i=0; i<N; i++)
	{
		assert(a[i] == 2*N + 1);
	}
	return 1;
}

File: tacas2020-benchmarks/vaphor/sina5.java
class vapexp{
	static int i,N;
	static int a[];
	static int b[];
	static long sum;

static void main()
{
	N=Support.random();

	sum = 0;
	i=0;
	while(i<N)
        {
                a[i] = 1;
		i=i+1;
        }

	i=0;
	while(i<N)
        {
                b[i] = 1;
		i=i+1;
        }

	i=0;
	while(i<N)
        {
		sum = sum + a[i];
		i=i+1;
        }

	i=0;
	while(i<N)
        {
		sum = sum + b[i];
		i=i+1;
        }

	i=0;
	while(i<N)
        {
		a[i] = a[i] + sum;
		i=i+1;
        }

	i=0;
	while(i<N)
        {
		assert(a[i] == 2*N+1);
		i=i+1;
	}
}

}
